File: support_tools/swigwin-3.0.12/Examples/test-suite/ruby/using_composition_runme.rb
#!/usr/bin/env ruby
#
# Put description here
#
# 
# 
# 
#

require 'swig_assert'

require 'using_composition'

include Using_composition

f = FooBar.new
if f.blah(3) != 3
  raise RuntimeError,"FooBar::blah(int)"
end

if f.blah(3.5) != 3.5
  raise RuntimeError,"FooBar::blah(double)"
end

if f.blah("hello") != "hello"
  raise RuntimeError,"FooBar::blah(char *)"
end


f = FooBar2.new
if f.blah(3) != 3
  raise RuntimeError,"FooBar2::blah(int)"
end

if f.blah(3.5) != 3.5
  raise RuntimeError,"FooBar2::blah(double)"
end

if f.blah("hello") != "hello"
  raise RuntimeError,"FooBar2::blah(char *)"
end


f = FooBar3.new
if f.blah(3) != 3
  raise RuntimeError,"FooBar3::blah(int)"
end

if f.blah(3.5) != 3.5
  raise RuntimeError,"FooBar3::blah(double)"
end

if f.blah("hello") != "hello"
  raise RuntimeError,"FooBar3::blah(char *)"
end

File: support_tools/swigwin-3.0.12/Examples/test-suite/python/using_composition_runme.py
from using_composition import *

f = FooBar()
if f.blah(3) != 3:
    raise RuntimeError, "FooBar::blah(int)"

if f.blah(3.5) != 3.5:
    raise RuntimeError, "FooBar::blah(double)"

if f.blah("hello") != "hello":
    raise RuntimeError, "FooBar::blah(char *)"


f = FooBar2()
if f.blah(3) != 3:
    raise RuntimeError, "FooBar2::blah(int)"

if f.blah(3.5) != 3.5:
    raise RuntimeError, "FooBar2::blah(double)"

if f.blah("hello") != "hello":
    raise RuntimeError, "FooBar2::blah(char *)"


f = FooBar3()
if f.blah(3) != 3:
    raise RuntimeError, "FooBar3::blah(int)"

if f.blah(3.5) != 3.5:
    raise RuntimeError, "FooBar3::blah(double)"

if f.blah("hello") != "hello":
    raise RuntimeError, "FooBar3::blah(char *)"
